lib/elixir/lib/kernel.ex

```diff
@@ -2292,8 +2292,18 @@ defmodule Kernel do
   In order to compare more than two clauses, the `cond/1` macro has to be used.
   """
   defmacro if(condition, clauses) do
-    do_clause = Keyword.get(clauses, :do, nil)
-    else_clause = Keyword.get(clauses, :else, nil)
+    {do_clause, clauses} = Keyword.pop_first(clauses, :do)
+    {else_clause, clauses} = Keyword.pop_first(clauses, :else)
+
+    case {do_clause, else_clause, clauses} do
```

Comment:
@lexmag Yes - I still have to validate `if true, []` and `if true, else: "foo"` I should have been clearer on this, sorry.